### PR TITLE
[Enhancement] Support show stats meta for external table

### PR DESCRIPTION
**Pull request by Youngwb**
Fixes #issue
#23327
1. support persist stats meta for external table
2. supoort show stats meta for external table

stats meta could refer to  https://docs.starrocks.io/zh-cn/latest/sql-reference/sql-statements/data-definition/SHOW%20META

eg:
```
mysql> show stats meta;
+---------------------------------------------------------+------------+---------+------+---------------------+------------+---------+
| Database                                                | Table      | Columns | Type | UpdateTime          | Properties | Healthy |
+---------------------------------------------------------+------------+---------+------+---------------------+------------+---------+
| test                                                    | detailDemo | ALL     | FULL | 2023-09-19 11:20:16 | {}         | 100%    |
| emr_test.ywb                                            | emps_unpar | ALL     | FULL | 2023-09-19 15:05:43 | {}         |         |
| emr_test.ywb                                            | emps       | ALL     | FULL | 2023-09-19 15:05:53 | {}         |         |
| emr_test.ywb                                            | emps_par   | ALL     | FULL | 2023-09-19 20:13:10 | {}         |         |
+---------------------------------------------------------+------------+---------+------+---------------------+------------+---------+
```
## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by stephen-shelby — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 116 / 128 (90.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|3|00.00%|[1167, 1168, 1169]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|20|25|80.00%|[111, 116, 124, 125, 130]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|8|9|88.89%|[2642]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeMgr.java|40|43|93.02%|[265, 743, 746]|
|:large_blue_circle:|com/starrocks/statistic/ExternalBasicStatsMeta.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java [hunk 1]
```diff
@@ -981,6 +982,25 @@ public static void loadJournal(GlobalStateMgr globalStateMgr, JournalEntity jour
                     globalStateMgr.getAnalyzeMgr().replayRemoveHistogramStatsMeta(histogramStatsMeta);
                     break;
                 }
+                case OperationType.OP_ADD_EXTERNAL_BASIC_STATS_META: {
+                    ExternalBasicStatsMeta basicStatsMeta = (ExternalBasicStatsMeta) journal.getData();
+                    globalStateMgr.getAnalyzeMgr().replayAddExternalBasicStatsMeta(basicStatsMeta);
+                    // The follower replays the stats meta log, indicating that the master has re-completed
+                    // statistic, and the follower's should refresh cache here.
+                    // We don't need to refresh statistics when checkpointing
+                    if (!GlobalStateMgr.isCheckpointThread()) {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java:991]: Better to add try/catch here to avoid start failure.

**Review by gengjun-git — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**